main.js

```diff
@@ -77,28 +79,28 @@ function deleteIdeaCard(ideaId) {
       var ideaToDelete = ideas[i];
       ideaToDelete.deleteFromStorage();
       ideas.splice(i, 1);
-      displayCards();
     }
   }
 }
 
-function favoriteIdeaCard() {
-
+function favoriteIdeaCard(ideaId) {
+  for (var i = 0; i < ideas.length; i++) {
+    if (ideaId === ideas[i].id) {
+      ideas[i].star = !ideas[i].star;
+      ideas[i].updateIdea();
+    }
+  }
```

Comment:
The favoriteIdeaCard function iterates through the ideas array and flips the star property of the appropriate idea. It calls the updateIdea method of that instance.